internal/stack/_static/logstash.conf.tmpl

```diff
@@ -21,14 +21,26 @@ filter {
 }
 
 output {
-  elasticsearch {
-    hosts => ["{{ $elasticsearch_host }}"]
-    user => '{{ fact "username" }}'
-    password => '{{ fact "password" }}'
-    ssl_enabled => true
-{{- if eq $elasticsearch_host "https://elasticsearch:9200" }}
-    ssl_certificate_authorities => "/usr/share/logstash/config/certs/ca-cert.pem"
-    document_id => "%{[@metadata][_ingest_document][id]}"
-{{- end }}
+  if [@metadata][_ingest_document][id] {
+    elasticsearch {
+      hosts => ["{{ $elasticsearch_host }}"]
+      user => '{{ fact "username" }}'
+      password => '{{ fact "password" }}'
+      ssl_enabled => true
+  {{- if eq $elasticsearch_host "https://elasticsearch:9200" }}
+      ssl_certificate_authorities => "/usr/share/logstash/config/certs/ca-cert.pem"
+      document_id => "%{[@metadata][_ingest_document][id]}"
+  {{- end }}
+    }
+  } else {
+    elasticsearch {
+      hosts => ["{{ $elasticsearch_host }}"]
+      user => '{{ fact "username" }}'
+      password => '{{ fact "password" }}'
+      ssl_enabled => true
+  {{- if eq $elasticsearch_host "https://elasticsearch:9200" }}
+      ssl_certificate_authorities => "/usr/share/logstash/config/certs/ca-cert.pem"
+  {{- end }}
+    }
```

Comment:
Does something like this not work? I have not tried this though. Just checking to see if we can avoid repeating the same lines and also looking at possibility of scaling with more conditions in future?

```suggestion
    elasticsearch {
      hosts => ["{{ $elasticsearch_host }}"]
      user => '{{ fact "username" }}'
      password => '{{ fact "password" }}'
      ssl_enabled => true
  {{- if eq $elasticsearch_host "https://elasticsearch:9200" }}
      ssl_certificate_authorities => "/usr/share/logstash/config/certs/ca-cert.pem"
  {{- end }}
  {{- if [@metadata][_ingest_document][id] }}
      document_id => "%{[@metadata][_ingest_document][id]}"
  {{- end }}
    }
```

Reply:
I'm all for less repeatable lines :-) but this approach doesn't work. As far as I understand the ```{{- if ``` directives are the template rending time directive. 
So for example for this
```
    elasticsearch {
      hosts => ["{{ $elasticsearch_host }}"]
      user => '{{ fact "username" }}'
      password => '{{ fact "password" }}'
      ssl_enabled => true
  {{- if eq $elasticsearch_host "https://elasticsearch:9200" }}
      ssl_certificate_authorities => "/usr/share/logstash/config/certs/ca-cert.pem"
  {{- end }}
  ```
  
  you will get the logstash configuration file on the disk that looks like this
  
  ```    
  elasticsearch {
      hosts => ["{{ $elasticsearch_host }}"]
      user => '{{ fact "username" }}'
      password => '{{ fact "password" }}'
      ssl_enabled => true
      ssl_certificate_authorities => "/usr/share/logstash/config/certs/ca-cert.pem"      
```

if `$elasticsearch_host` equals `"https://elasticsearch:9200"` in this case.

We need the runtime condition here instead that happens during the logstash pipeline execution.

Reply:
Ohh yes.. I missed to notice that...